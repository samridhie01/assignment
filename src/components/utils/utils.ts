import { PlanetData } from "../types/types";

export const  comparePlanets = function(data:PlanetData):boolean {
  let foundElem = false;
 if (localStorage.getItem("favs") != null) {
    const addedFavs = JSON.parse(localStorage.getItem("favs")!);
    addedFavs.forEach((element: PlanetData) => {
      if (element.name === data.name) {
        foundElem=  true;
      }
    });
  }
  return foundElem;
};

export const getSearchKey =()=>{

    return (localStorage.getItem("searchKey") || "");
};

export const getPageNum =()=>{
    return localStorage.getItem("page");
};

export const setPageNum = (page:string) =>{
    localStorage.setItem("page", page);
}
