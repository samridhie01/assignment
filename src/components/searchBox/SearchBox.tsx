import React, { useState } from "react";
import {
  Button,
  Box,
  TextField,
  Theme,
  makeStyles,
  createStyles
} from "@material-ui/core";
import {getSearchKey} from "../utils/utils";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    alignCenter:{
      alignSelf: "center",
      width: "50%"
    },
    customStyle:{
     width: 350,
      padding: "20px"
    }
  })
);

interface Search {
  handleSearch: () => void;
}


const SearchBox: React.FC<Search> = ({ handleSearch }) => {
  const [searchItem, setSearchItem] = useState<string>("");
  const handleSearchChange = (event: any) => {
    setSearchItem(event.target.value);
    localStorage.setItem("searchKey", event.target.value);
  };

  const handleSearchButtonClick =() =>{
    handleSearch();
  }

  const classes = useStyles();

  return (
    <>
      <div >
      <Box m={4} border={1} width={400} height={150} display="flex" flexDirection={"column"} data-testid={"search-section"}>
        <TextField
          variant="outlined"
          onChange={handleSearchChange}
          placeholder="Enter climate to filter planets"
          className={classes.customStyle}
          fullWidth={false}
          inputProps={{ "data-testid": "climate-filter" }}
          value={getSearchKey() || ''}
        />
        <Button
          variant="contained"
          onClick={handleSearchButtonClick}
          fullWidth={false}
          data-testid={'searchBtn'}
          className={classes.alignCenter}
        >
          Search
        </Button>
        </Box>
      </div>
    </>
  );
};

export default SearchBox;
