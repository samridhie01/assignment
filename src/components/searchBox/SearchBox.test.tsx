import SearchBox from "./SearchBox";
import { render, getByTestId, fireEvent } from '@testing-library/react';
import React from "react";
import {getSearchKey} from "../utils/utils";


describe("Search Box", ()=>{

    const renderComponent = () => render(
        <SearchBox handleSearch={jest.fn} />
    );

    it('should render snapshot', () => {
        const { asFragment } = render(
            <SearchBox handleSearch={jest.fn} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should handle search button', () => {
        const filterName= "arid";
        const {getByTestId} = renderComponent();
        const searchTxt = getByTestId("climate-filter")
        expect(searchTxt).toBeInTheDocument();
        fireEvent.change(searchTxt, {target: {value :filterName}});
        expect(getSearchKey()).toEqual(filterName)
    });
})

