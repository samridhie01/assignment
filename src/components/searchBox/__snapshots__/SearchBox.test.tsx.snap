// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root MuiBox-root-3"
      data-testid="search-section"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-customStyle-2"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="climate-filter"
            placeholder="Enter climate to filter planets"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-5"
              style="width: 0.01px;"
            >
              <span>
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-alignCenter-1"
        data-testid="searchBtn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
